File: KHost.App/ClientApp/src/app/models/QueuedSong.ts
import { ModelWithId } from "./ModelWIthId";
import { ModelWithPosition } from "./ModelWithPosition";
import { QueuedSinger } from "./QueuedSinger";
import { Song } from "./Song";

export class QueuedSong implements ModelWithId, ModelWithPosition {

    id?: number;

    songId?: number;

    queuedSingerId?: number;

    song?: Song;

    queuedSinger?: QueuedSinger

    position: number = 0;
}

File: KHost.Abstractions/Models/QueuedSong.cs
namespace KHost.Abstractions.Models
{
    public class QueuedSong : BaseModel, IModelWithPosition, IModelWithId
    {
        public int? Id { get; set; }

        public int SongId { get; set; }

        public int QueuedSingerId { get; set; }

        public int Position { get; set; }

        public override QueuedSong Clone() => (base.Clone() as QueuedSong)!;
    }
}
